osquery/events/events.cpp

```diff
@@ -277,6 +282,46 @@ void EventSubscriberPlugin::expireIndexes(
   db->Put(kEvents, index_key + "." + list_type, new_indexes);
 }
 
+void EventSubscriberPlugin::expireCheck() {
+  auto db = DBHandle::getInstance();
+  auto data_key = "data." + dbNamespace();
+  auto eid_key = "eid." + dbNamespace();
+
+  std::vector<std::string> keys;
+  db->ScanPrefix(kEvents, keys, data_key);
+  if (keys.size() <= FLAGS_events_max) {
+    return;
+  }
+
+  // There is an overflow of events buffered for this subscriber.
```

Comment:
At this point a `WARN`-level log should be generated to indicate the number of events about to be expired.